### PR TITLE
remove mac promo experiment

### PR DESCRIPTION
**Pull request by brindy**
Task/Issue URL: https://app.asana.com/0/0/1204731397004150/f
Tech Design URL:
CC:

BSK: https://github.com/duckduckgo/BrowserServicesKit/pull/401

**Description**:

Removes Mac promo experiment while adding share link/text support to the RMF

**Steps to test this PR**:
1. Clean install, run the app and get through onboarding.
2. Restart the app, a sample message should display showing an RMF message.
3. Check RMF pixels fire as expected
4. Using the share sheet should *NOT* dismiss the RMF

**Copy Testing**:

* [ ] Use of correct apostrophes in new copy, ie `’` rather than `'`

**Orientation Testing**:

* [ ] Portrait
* [ ] Landscape

**Device Testing**:

* [ ] iPhone SE (1st Gen)
* [ ] iPhone 8
* [ ] iPhone X
* [ ] iPhone 14 Pro
* [ ] iPad

**OS Testing**:

* [ ] iOS 14
* [ ] iOS 15
* [ ] iOS 16

**Theme Testing**:

* [ ] Light theme
* [ ] Dark theme

---
###### Internal references:
[Software Engineering Expectations](https://app.asana.com/0/59792373528535/199064865822552)
[Technical Design Template](https://app.asana.com/0/59792373528535/184709971311943)

### CODE REVIEW

**Comment by amddg44** [comment]
@brindy the change to triggering the pixel shown events in onAppear is looking great! One issue I've spotted though - the remote message doesn't disappear when tapping the secondary button (or when tapping the primary buttons with non share actions)

**Comment by brindy** [comment]
@amddg44 thanks - that's fixed now.

File: DuckDuckGoTests/MacPromoExperimentTests.swift
Comment (amddg44) [inline DuckDuckGoTests/MacPromoExperimentTests.swift:1]: @brindy I think you meant to delete this file (not just the contents?) 🙂

**Review by amddg44 — APPROVED** [review]
LGTM!

**Pull request merged**